packages/react/src/utils/appearance.ts

```diff
@@ -0,0 +1,45 @@
+import { createRoot } from 'react-dom/client';
+import type { Appearance as JsAppearance, IconRenderer, IconOverrides as JsIconOverrides, IconKey } from '@novu/js/ui';
+
+import type { ReactAppearance, ReactIconRenderer } from './types';
+
+function adaptIconRenderer(reactRenderer: ReactIconRenderer): IconRenderer {
+  return (el: HTMLDivElement, props: { class?: string }) => {
+    const root = createRoot(el);
+
+    root.render(reactRenderer(props ?? {}));
+
+    return () => {
+      setTimeout(() => root.unmount(), 0);
+    };
+  };
+}
```

Comment:
The biggest downside of this will be that these components won't react to the state change that is happening in the parent application

Reply:
Do we have some other uses that we have that allow it?